From f42b4c30d7d13f7ec37718928b13a064936b6937 Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Sat, 3 Nov 2018 04:12:11 +0100
Subject: [PATCH 01/25] Add define for O_TMPFILE

Until we have a proper backport of the feature, this seems
to do the trick for fixing build issues in Android Pie.

Change-Id: I4a0f87f912a8fa9a2367e22f7a7b43a18551768c
Signed-off-by: Kevin F. Haggerty <haggertk@lineageos.org>
---
 include/asm-generic/fcntl.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/asm-generic/fcntl.h b/include/asm-generic/fcntl.h
index b0e7e947fb1e..8e6043e3aa5c 100644
--- a/include/asm-generic/fcntl.h
+++ b/include/asm-generic/fcntl.h
@@ -84,6 +84,14 @@
 #define O_PATH		010000000
 #endif
 
+#ifndef __O_TMPFILE
+#define __O_TMPFILE	020000000
+#endif
+
+/* a horrid kludge trying to make sure that this will fail on old kernels */
+#define O_TMPFILE (__O_TMPFILE | O_DIRECTORY)
+#define O_TMPFILE_MASK (__O_TMPFILE | O_DIRECTORY | O_CREAT)
+
 #ifndef O_NDELAY
 #define O_NDELAY	O_NONBLOCK
 #endif
-- 
2.17.1

